components/button/button-add.js

```diff
@@ -0,0 +1,131 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { labelStyles } from '../typography/styles.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component for quickly adding items to a specific locaiton.
+ */
+class ButtonAdd extends FocusMixin(LocalizeCoreElement(LitElement)) {
+	static get properties() {
+		return {
+			/**
+			 * The text to be shown in the tooltip as a hint when visible-text is false
+			 * @type {string}
+			 */
+			label: { type: String },
+			/**
+			 * The text to show in the button when visible-text is true
+			 * @type {string}
+			 */
+			text: { type: String },
+			/**
+			 * When true, show the button with icon and visible text. When false, only show icon.
+			 * @type {boolean}
+			 */
+			visibleText: { type: Boolean, reflect: true, attribute: 'visible-text' }
+		};
+	}
+
+	static get styles() {
+		return [labelStyles, css`
+			:host {
+				--d2l-button-add-line-style: solid;
+			}
+			button {
+				align-items: center;
+				background-color: transparent;
+				border: 0;
+				box-shadow: none;
+				display: flex;
+				font-family: inherit;
+				justify-content: center;
+				outline: none;
+				padding: 0;
+				position: relative;
+				white-space: nowrap;
+				width: 100%;
+			}
+
+			.line {
+				border-top: 1px var(--d2l-button-add-line-style) var(--d2l-color-mica);
+				margin: 3px 0; /** hover/click target */
+				width: 100%;
+			}
+			button:hover .line,
+			button:focus .line {
+				border-top-color: var(--d2l-color-celestine);
+			}
+
+			.content {
+				align-items: center;
+				background-color: white;
+				display: flex;
+				position: absolute;
+			}
+			:host([visible-text]) .content {
+				color: var(--d2l-color-celestine);
+				height: 1.5rem;
+				padding: 0 0.3rem;
+			}
+
+			:host([visible-text]) d2l-icon,
+			:host(:not([visible-text])) button:hover d2l-icon,
+			:host(:not([visible-text])) button:focus d2l-icon {
+				color: var(--d2l-color-celestine);
+			}
+			:host(:not([visible-text])) d2l-icon {
+				color: var(--d2l-color-galena);
+				margin: -3px; /** hover/click target */
+				padding: 3px; /** hover/click target */
+			}
+			:host([visible-text]) d2l-icon {
+				padding-inline-end: 0.2rem;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+
+		this.visibleText = false;
+		this._buttonId = getUniqueId();
+
+		this.addEventListener('click', this._onClick);
+	}
+
+	static get focusElementSelector() {
+		return 'button';
+	}
+
+	render() {
+		const text = this.text || this.localize('components.button-add.addItem');
+		const label = this.label || this.localize('components.button-add.addItem');
+
+		return html`
+			<button class="d2l-label-text" id="${this._buttonId}">
```

Comment:
I don't know how I feel about using label styles for non-label things... I could see us want to change these in the future and regret applying them to places like this.